src/main/python/app.py

```diff
@@ -1,130 +1,80 @@
         self.on_dashboard_clicked()
 
-    def _init_geometry(self):
-        # window size
-        set_base_geometry(self, 1024, 720, fixed=True)
-
-        # hide title bar
-        # self.setWindowFlags(Qt.FramelessWindowHint)
-
-    def _init_ui(self):
-        self.setObjectName("App")
+    def update_account(self):
+        # fetch account information
+        self._api_get_call()
 
-        sidebar_width = 100
-        navigation_height = 45
+        # update navigation credit value
+        self.navigation.set_credit(self.total_balance)
 
-        window = QFrame(self)
-        self.setCentralWidget(window)
+        # update account username and credit value
+        self.account.update_info(self.username, self.total_balance)
 
-        # side bar
-        self.sidebar = SideBar(window, self.cxt)
-        self.sidebar.setFixedSize(sidebar_width, self.height())
-        self.sidebar.move(0, 0)
+    def on_dashboard_clicked(self):
+        self.sidebar.on_dashboard_clicked()
+        self._sidebar_widget_updated(Dashboard)
 
-        # top navigation
-        self.navigation = Navigation(window, self.cxt)
-        self.navigation.setFixedSize(self.width()-sidebar_width, navigation_height)
-        self.navigation.move(sidebar_width, 0)
+    def on_resources_clicked(self):
+        self.sidebar.on_resources_clicked()
+        self._sidebar_widget_updated(Resources)
 
-        self.main_window = MainWindow(window, self.cxt)
-        self.main_window.setFixedSize(self.width()-sidebar_width, self.height()-navigation_height)
-        self.main_window.move(sidebar_width, navigation_height)
+    def on_jobs_clicked(self):
+        self.sidebar.on_jobs_clicked()
+        self._sidebar_widget_updated(Jobs)
 
-        # mask & right account bar
-        self.mask = Mask(window, self.cxt)
-        self.mask.setFixedSize(self.width(), self.height())
-        self.mask.move(self.width(), 0)
-        self.mask.clicked_area.setFixedSize(self.width(), self.height())
-        self.mask.clicked_area.move(0, 0)
+    def on_settings_clicked(self):
+        self.sidebar.on_settings_clicked()
+        self._sidebar_widget_updated(Settings)
 
-        self.account = Account(self.mask, self.cxt)
-        self.account.setFixedSize(200, self.height())
-        self.account.move(self.mask.width()-200, 0)
+    def on_notification_clicked(self):
+        popup = Notification(self.cxt)
+        popup.exec_()
 
-        # connect function
-        self.sidebar.dashboard.clicked.connect(self.on_dashboard_clicked)
-        self.sidebar.resources.clicked.connect(self.on_resources_clicked)
-        self.sidebar.jobs.clicked.connect(self.on_jobs_clicked)
-        self.sidebar.settings.clicked.connect(self.on_settings_clicked)
+    # credit history popup
+    def on_credit_history_clicked(self):
+        popup = CreditHistory(self.cxt)
+        popup.exec_()
 
-        self.navigation.menu_button.clicked.connect(self.on_menu_clicked)
+    def on_about_clicked(self):
```

Comment:
I was planning to remove it later on. Since this is a just PR to refactor the project, I was thinking we should just make the codebases to be under the new structure without changing too many thing. In addition, we are talking about an option on user's account menu, and it won't be good place to point the user to our official website right there. And we have already got one within the setting page.

Reply:
good call - maybe this isn't the right place to put a link. However, what purpose does the about buttons serve at this moment?

Reply:
To be honest, I forgot. It's been there a long time, and since it's our primary focus overall, we just have not time to take of this sidebar. Maybe I should just comment it out this time.